modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/RexImpTable.java

```diff
@@ -3077,6 +3077,19 @@ private static class BinaryImplementor extends AbstractRexCallImplementor {
         argValueList = FlatLists.append(argValueList, fieldComparator);
       }
 
+      if (type0 == BigDecimal.class && type1 == BigDecimal.class && op == DIVIDE) {
+        int precision = call.getType().getPrecision();
+        int scale = call.getType().getScale();
+
+        assert precision != RelDataType.PRECISION_NOT_SPECIFIED || scale != RelDataType.SCALE_NOT_SPECIFIED
```

Comment:
we can call the cast : 
SELECT CAST(1 AS DECIMAL(32767, 32767))
but we failed with:
SELECT CAST(1 AS DECIMAL(32, 32))
seems the root of the problem is somewhere here

Reply:
The issue with DECIMAL(32767, 32767) is going to be addressed here:
https://issues.apache.org/jira/browse/IGNITE-23172